src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax;
+
+public abstract class Figure implements AreaCalculator, Drawable {
+    protected String color;
+    private ColorSupplier colorSuppliers;
+
+    public Figure(ColorSupplier colorSuppliers) {
+        this.colorSuppliers = colorSuppliers;
```

Comment:
Figure class shouldn't have colorSupplier dependency. Consider passing color to the Figure directly.